zstacklib/zstacklib/utils/file_system_upload_task.py

```
@@ -0,0 +1,85 @@
+        file_obj.seek(slice_offset)
+        return file_obj
+
+    def create_file_if_not_exists(self):
+        if self.task_created:
+            return
+
+        with lock.NamedLock("upload-file-task-%s" % self.taskUuid):
+            if not self.task_created:
+                r, _, e = bash_roe("fallocate -l %s %s" % (self.expectedSize, self.installPath))
+                if r != 0:
+                    raise Exception("Failed to allocate file space for %s, because %s " % (self.installPath, str(e)))
```

Ruff (0.14.10)

Create your own exception

(TRY002)